lib/integrations/awsoidc/opentunnelec2_keys.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsoidc
+
+import (
+	"context"
+	"crypto/ed25519"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2instanceconnect"
+	"github.com/gravitational/trace"
+	"golang.org/x/crypto/ssh"
+
+	"github.com/gravitational/teleport/lib/auth/native"
+	"github.com/gravitational/teleport/lib/modules"
+)
+
+// generatePrivatePublicKey returns a public and its private key usable for creating an [ssh.Signer].
+func generatePrivatePublicKey() (publicKey any, privateKey any, err error) {
+	if modules.GetModules().IsBoringBinary() {
+		privKey, err := native.GeneratePrivateKey()
+		if err != nil {
+			return nil, nil, trace.Wrap(err)
+		}
+
+		return privKey.Public(), privKey, nil
+	}
+
+	pubKey, privKey, err := ed25519.GenerateKey(nil)
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
```

Comment:
I'm not a fan of hard-coding algorithm selection in packages that are ultimately not crypto-focused, at least not if we can help it.  Too easy to overlook.  Could we instead create a helper/option in `native` for the "small keys preferred" case? (e.g. `native.GenerateEphemeralPrivateKey()`, `native.GeneratePrivateKey(native.PreferSmallerKeys(true))`, etc)

Reply:
Looks good, thank you :+1:

Reply:
I bumped into a dependency cycle, but should be easy to fix